handlers/handlers.go

```diff
@@ -1,194 +1,127 @@
 package handlers
 
 import (
-	"time"
+	"patches/models"
+
 	"encoding/json"
 	"log"
 	"net/http"
-	"database/sql"
 	"io/ioutil"
+	"fmt"
+	"time"
 
-    _ "github.com/go-sql-driver/mysql"
-	_ "github.com/mattn/go-sqlite3"
+	"github.com/gorilla/schema"
 )
 
-type Patch struct {
-	Timestamp   time.Time      		 `json:"timestamp,omitempty"`
-	Patch 		string 				 `json:"patch,omitempty"`
-	Convo_ID    string               `json:"convo_id,omitempty"`
-	User_ID     string         		 `json:"user_id,omitempty"`
-	Type 		string 				 `json:"type,omitempty"`
+type Env struct {
+	DB models.Datastore
 }
 
-// Connect to database
-func dbConnect() (*sql.DB, error) {
-	db, err := sql.Open("sqlite3", "../patches.db")
-	if err != nil {
-		log.Fatal(err)
-		return db, err
-	}
-	// test connection
-	err = db.Ping()
-	if err != nil {
-		log.Fatal(err)
-		return db, err
-	}
-	return db, nil
+type Filter struct {
+    Conversation   	[]string 		`schema:"convoID"`
+    User 			[]string 		`schema:"userID"`
+    Type  			[]string 		`schema:"type"`
+	MaxTime     	time.Time 		`schema:"maxTime"`	//Newest
+	MinTime     	time.Time 		`schema:"minTime"`	//Oldest
 }
 
-// Create new patch in database
-func PostPatchesHandler(w http.ResponseWriter, r *http.Request) {
+// Read all patches from database
+func (env *Env)GetPatchesHandler(w http.ResponseWriter, r *http.Request) {
+	if err := r.ParseForm(); err != nil {
+        // Handle error
+    }
 
-	// Read http request
-	body, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		log.Print("Error reading request")
-		log.Print(err)
+    filter := new(Filter)
+    if err := schema.NewDecoder().Decode(filter, r.Form); err != nil {
+        // Handle error
+    }
+	
+	// Create filter string
+	filterString := ""
+	for _, convoID := range filter.Conversation {
+        filterString = fmt.Sprintf("%s%s%s", filterString, " AND convo_id = ", convoID) 
```

Comment:
If you used the `strings.Builder` method here, it would look like the following. This goes for the rest of the `filterString` stuff.
```suggestion
        fmt.Fprintf(filterString, " AND convo_id = %s", convoID) 
```